src/core/bonded_interactions/angle_cosine.hpp

```diff
@@ -31,46 +31,38 @@
 #include "utils.hpp"
 
 #ifdef BOND_ANGLE
+#include "angle_common.hpp"
 #include "grid.hpp"
 
 /** set parameters for the angle potential.
-
-    \todo The type of the angle potential
-    is chosen via config.hpp and cannot be changed at runtime.
-*/
+ *
+ *  \todo The type of the angle potential
+ *  is chosen via config.hpp and cannot be changed at runtime.
+ */
 int angle_cosine_set_params(int bond_type, double bend, double phi0);
 
 /************************************************************/
 
-/** Computes the three body angle interaction force and adds this
-    force to the particle forces.
-    @param p_mid     Pointer to second/middle particle.
-    @param p_left    Pointer to first/left particle.
-    @param p_right   Pointer to third/right particle.
-    @param iaparams  bond type number of the angle interaction.
-    @param force1 returns force of particle 1
-    @param force2 returns force of particle 2
-    @return 0
-*/
-inline int calc_angle_cosine_force(Particle *p_mid, Particle *p_left,
-                                   Particle *p_right,
-                                   Bonded_ia_parameters *iaparams,
+/** Computes the three-body angle interaction force.
+ *  @param[in]  p_mid     Second/middle particle.
+ *  @param[in]  p_left    First/left particle.
+ *  @param[in]  p_right   Third/right particle.
+ *  @param[in]  iaparams  Bonded parameters for the angle interaction.
+ *  @param[out] force1    Force on particle 1.
+ *  @param[out] force2    Force on particle 2.
+ *  @retval 0
+ */
+inline int calc_angle_cosine_force(Particle const *p_mid,
```

Comment:
please also use ``Vector3d`` and return instead of using output variables.

Reply:
Do you mean `/src/core/forces_inline.hpp` should take the following C++11 form?
```c++
// current
bond_broken = calc_umbrella_pair_force(p1, p2, iaparams, dx, force);
// C++11
std::tie(bond_broken, force) = calc_umbrella_pair_force(p1, p2, iaparams, dx);
// C++17
auto [bond_broken, force] = calc_umbrella_pair_force(p1, p2, iaparams, dx);
```

Reply:
yes

Reply:
I'll take care of that in another PR, as it requires updating the functions for pressure and non-bonded forces. Noted in #2502

Reply:
:+1: